module/User/Module.php

```diff
@@ -168,6 +168,10 @@ public function getServiceConfig()
                     );
                 },
 
+                'user_remoteaddress' => function ($sm) {
+                    $remote = new \Zend\Http\PhpEnvironment\RemoteAddress();
+                    return $remote->getIpAddress();
+                },
```

Comment:
Zend framework's ability to add abstraction layers to every single php function or global variable still amazes me every time.